torch/_inductor/overrides.py

```diff
@@ -35,21 +35,25 @@ def __torch_function__(self, func, types, args=(), kwargs=None):
             config.fallback_random
             and replacements[func] in replacements_using_triton_random
         ):
-            return replacements[func](*args, **kwargs)
+            if func != torch.nn.functional.dropout or (
+                func == torch.nn.functional.dropout
+                and args[0].device != torch.device("cpu")
+            ):
+                return replacements[func](*args, **kwargs)
         return func(*args, **kwargs)
 
 
 patch_functions = AutogradMonkeypatch
 
 
-def replace_fx(gm: torch.fx.GraphModule):
+def replace_fx(gm: torch.fx.GraphModule, example_inputs):
     # Sometimes patch_functions() misses things already in the graph
     for node in reversed(list(gm.graph.nodes)):
         if node.op == "call_function" and node.target in replacements:
             if (
                 config.fallback_random
-                and replacements[node.target] in replacements_using_triton_random
-            ):
+                or example_inputs[0].device == torch.device("cpu")
```

Comment:
safer to check if all example_inputs are on cpu? similar to `fuse_fx`
```python
def fuse_fx(gm: torch.fx.GraphModule, example_inputs):
    is_cpu = all(
        example_input.device == torch.device("cpu")
        for example_input in example_inputs
        if isinstance(example_input, torch.Tensor)
    )
```

Reply:
Changed in [7aa97a7](https://github.com/pytorch/pytorch/pull/97002/commits/7aa97a7d8a8bcc3dd7cf40eaad52223cd3eebc45)